src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java

```diff
@@ -143,29 +143,42 @@ private DocumentLink findDocumentByFileName(String fileName, SscsCaseData sscsCa
         List<DwpDocument> dwpDocuments = sscsCaseData.getDwpDocuments();
 
         if (isNotEmpty(dwpDocuments)) {
-            var result = dwpDocuments.stream()
-                .filter(document -> fileName.equals(document.getValue().getDocumentFileName()))
-                .findAny()
-                .orElse(null);
-
-            if (result != null) {
-                return result.getValue().getDocumentLink();
+            DocumentLink dwpLink =  dwpDocuments.stream()
+                .map(dwpDocument -> {
+                    if (fileName.equals(dwpDocument.getValue().getDocumentFileName())) {
+                        return dwpDocument.getValue().getDocumentLink();
+                    } else if (dwpDocument.getValue().getEditedDocumentLink() != null
+                        && fileName.equals(dwpDocument.getValue().getEditedDocumentLink().getDocumentFilename())) {
+                        return dwpDocument.getValue().getEditedDocumentLink();
+                    }
+                    return DocumentLink.builder().build();
+                })
+                .filter(documentLink -> documentLink.getDocumentUrl() != null)
+                .findAny().orElse(null);
+            if (dwpLink != null) {
+                return dwpLink;
             }
         }
 
         List<SscsDocument> sscsDocuments = sscsCaseData.getSscsDocument();
 
         if (isNotEmpty(sscsDocuments)) {
-            var doc = sscsDocuments.stream()
-                .filter(d -> fileName.equals(d.getValue().getDocumentFileName()))
-                .findAny()
-                .orElse(null);
-
-            if (doc != null) {
-                return doc.getValue().getDocumentLink();
+            DocumentLink sscslink =  sscsDocuments.stream()
+                .map(sscsDocument -> {
+                    if (fileName.equals(sscsDocument.getValue().getDocumentFileName())) {
+                        return sscsDocument.getValue().getDocumentLink();
+                    } else if (sscsDocument.getValue().getEditedDocumentLink() != null
+                        && fileName.equals(sscsDocument.getValue().getEditedDocumentLink().getDocumentFilename())) {
+                        return sscsDocument.getValue().getEditedDocumentLink();
+                    }
+                    return DocumentLink.builder().build();
+                })
+                .filter(documentLink -> documentLink.getDocumentUrl() != null)
+                .findAny().orElse(null);
+            if (sscslink != null) {
+                return sscslink;
```

Comment:
can it extracted to a single method and called for both dip documents and sscs documents ?